### DISCUSSION TITLE
Dependency usage during initialization

### DISCUSSION THREAD

**Discussion by chrsmys** [pointfreeco/swift-dependencies#157] [Q&A] [answered]
I've been using swift dependencies for a bit now and one topic I didn't find covered (apologies if I missed it!) is the best way to handle dependency usage during initialization. The problem specifically being that you can't access self until after init. Below is a simplified example to illustrate what I would like to be able to do but can't:
```
struct ButtonTappedEvent {
   @Dependency(\.uuid) var uuid
   private let sessionID: UUID

   init() {
       sessionID = uuid() // self accessed before initialization error
   }
}
```

A workaround for this that I have been using is something along these lines:
```
struct ButtonTappedEvent {
   struct Dependencies {
       @Dependency(\.uuid) var uuid
   }
   private let sessionID: UUID

   init(dependencies: Dependencies = .init) {
       sessionID = dependencies.uuid()
   }
}
```

I'm curious if anyone else has come up with a better solution or if there is something I am missing here. Thanks!

**Comment by jshier** [comment, marked as answer]
You can access the dependency right in the initializer:

```swift
init(dependencies: Dependencies = .init) {
  @Dependency(\.uuid) var uuid
  sessionID = uuid()
}
```

**Reply by chrsmys** [reply]
Wow, for some reason I assumed they had to be instance variables and couldn't be local variables. I guess that assumption stemmed from all of the example usage I've seen. Thanks for the simple solution here!

**Reply by gsabran** [reply]
This is really great.
